mineinabyss-features/src/main/kotlin/com/mineinabyss/armorstandlock/ArmorStandLockingListener.kt

```diff
@@ -92,4 +99,114 @@ class ArmorStandLockingListener : Listener {
             isCancelled = true
         }
     }
+
+    @EventHandler
+    fun EntityMoveEvent.onEntityMove() {
+        /**
+         * This listener is used to prevent gravity and water from affecting armor stands.
+         * This is done by setting the entity's velocity to 0 and cancelling the event.
+         */
+        if (entityType != EntityType.ARMOR_STAND) return
+
+        val armorStand = entity.toGeary().get<LockArmorStand>() ?: return
+
+        if (!armorStand.lockState) return
+
+        // We only care about ArmorStands that have fishing hooks as passengers
+        if (entity.passengers.none { it.type == EntityType.FISHING_HOOK }) return
+
+        entity.velocity = Vector(0.0, 0.0, 0.0)
+        isCancelled = true
+    }
+
+    @EventHandler
+    fun PlayerFishEvent.onFishArmorstand() {
+        /**
+         * This listener is used to prevent players from fishing armor stands if they are not allowed to.
+         * This is done by simply cancelling the event. A player can fish an armor stand if they are allowed to via /mia lock
+         * or if they have the permission to bypass the lock.
+         */
+        if (state != PlayerFishEvent.State.CAUGHT_ENTITY) return // We only care about catching entities
+
+        if (caught == null) return // We only care about non-null caught entities
+        if (caught!!.type != EntityType.ARMOR_STAND) return // We only care about armor stands
+
+        val armorStandEntity = caught as ArmorStand
```

Comment:
Since both minecraft and geary use the term entity, I would call this `armorStand` and the other `armorStandGeary`, or call this one `armorStandBukkit`. That matches the existing `BukkitEntity` typealias that we use.